Source/Core/InputCommon/InputConfig.cpp

```diff
@@ -117,3 +117,17 @@ bool InputConfig::ControllersNeedToBeCreated() const
 {
 	return m_controllers.empty();
 }
+
+bool InputConfig::IsControllerControlledBySomeTypeOfGamepadDevice(int index) const
+{
+	if (static_cast<size_t>(index) >= m_controllers.size())
+		return false;
+
+	const auto& controller = m_controllers.at(index).get()->default_device;
+
+	// Filter out anything which obviously not a gamepad
+	return !((controller.source == "Keyboard") // OSX Keyboard/Mouse
```

Comment:
> (assuming I did the inversions correctly):

That's always the problem. Just relax and let the compiler do the inversions for us.